src/cpp/transport/TCPTransportInterface.cpp

```diff
@@ -778,55 +655,58 @@ void TCPTransportInterface::perform_rtcp_management_thread(TCPChannelResource *p
-            << p_channel_resource->socket()->remote_endpoint().address() << ":" \
-            << p_channel_resource->socket()->remote_endpoint().port() << ")");
+    logInfo(RTCP, "START perform_rtcp_management_thread " << IPLocator::toIPv4string(channel->locator()) \
+            << ":" << IPLocator::getPhysicalPort(channel->locator()) << " (" \
+            << channel->socket()->local_endpoint().address() << ":" \
+            << channel->socket()->local_endpoint().port() << "->" \
+            << channel->socket()->remote_endpoint().address() << ":" \
+            << channel->socket()->remote_endpoint().port() << ")");
 */
-    while (p_channel_resource->alive())
+    while (channel && TCPChannelResource::TCPConnectionStatus::TCP_CONNECTED == channel->tcp_connection_status())
     {
-        if (p_channel_resource->connection_established())
+        if (channel->connection_established())
         {
             // KeepAlive
             if (config->keep_alive_frequency_ms > 0 && config->keep_alive_timeout_ms > 0)
             {
                 time_now = std::chrono::system_clock::now();
 
                 // Keep Alive Management
-                if (!p_channel_resource->waiting_for_keep_alive_ && time_now > next_time)
+                if (!channel->waiting_for_keep_alive_ && time_now > next_time)
                 {
-                    std::unique_lock<std::mutex> scopedLock(sockets_map_mutex_);
-                    rtcp_message_manager_->sendKeepAliveRequest(p_channel_resource);
-                    p_channel_resource->waiting_for_keep_alive_ = true;
+                    std::unique_lock<std::mutex> scopedLock(sockets_map_mutex_); // Why mutex here?
+                    rtcp_message_manager_->sendKeepAliveRequest(channel);
+                    channel->waiting_for_keep_alive_ = true;
                     next_time = time_now + std::chrono::milliseconds(config->keep_alive_frequency_ms);
                     timeout_time = time_now + std::chrono::milliseconds(config->keep_alive_timeout_ms);
                 }
-                else if (p_channel_resource->waiting_for_keep_alive_ && time_now >= timeout_time)
+                else if (channel->waiting_for_keep_alive_ && time_now >= timeout_time)
                 {
                     // Disable the socket to erase it after the reception.
-                    close_tcp_socket(p_channel_resource);
-                    break;
+                    close_tcp_socket(channel);
+                    channel.reset();
                 }
             }
         }
+
+        //TODO Remove
         eClock::my_sleep(100);
     }
-    logInfo(RTCP, "End perform_rtcp_management_thread " << p_channel_resource->locator());
+    logInfo(RTCP, "End perform_rtcp_management_thread " << channel->locator());
 }
 
-void TCPTransportInterface::perform_listen_operation(TCPChannelResource *p_channel_resource)
+void TCPTransportInterface::perform_listen_operation(
+        std::shared_ptr<TCPChannelResource> channel)
```

Comment:
According to the style guide, one-parameter methods should be declared in one line if doesn't surpass the limit of 120 characters.